pip_audit/_state.py

```diff
@@ -98,6 +102,100 @@ def finalize(self) -> None:
         raise NotImplementedError  # pragma: no cover
 
 
+class StatusLog:
+    """
+    Displays a status indicator with an optional log panel to display logs
+    for external processes.
+
+    This code is based off of Rich's `Status` component:
+        https://github.com/Textualize/rich/blob/master/rich/status.py
+    """
+
+    LOG_PANEL_HEIGHT = 10
```

Comment:
This isn't perfect. The actual height of the panel is 10 characters and I'm getting the last 10 lines of the log. However, the panel doesn't display the last 10 lines if some of the lines are long enough to wrap in the panel. I'll have a look online and see what I can do about that.

Reply:
Making sure I understand: does that mean that the panel will "skip" some lines, since some long lines wrap and effectively push the lines that follow out of sight?

Reply:
Yeah that's exactly right. We might not necessarily see the last few lines in the log because of this.

Reply:
Yeah, I think that's an acceptable trade-off for now. Let's leave a `NOTE`, but otherwise IMO we shouldn't consider this a blocking issue.

Reply:
[3196a16](https://github.com/pypa/pip-audit/pull/523/commits/3196a16c67fcac0475c0b8fb03eaf3d24d0802b9)